Marlin/src/lcd/dogm/ultralcd_DOGM.cpp

```diff
@@ -374,31 +374,43 @@ void MarlinUI::clear_lcd() { } // Automatically cleared by Picture Loop
                         one_chr_width = MENU_FONT_WIDTH;
     #endif
 
-    // Center the label and value lines on the middle line
-    uint8_t baseline = extra_row ? (LCD_PIXEL_HEIGHT) / 2 - 1
-                                 : (LCD_PIXEL_HEIGHT + EDIT_FONT_ASCENT) / 2;
+    // Center the label and value1 lines on the middle line
+    uint8_t baseline = extra_row ? (LCD_PIXEL_HEIGHT) / (2 + (value2 == NULL ? 0 : 1)) - 1
+                                 : (LCD_PIXEL_HEIGHT + EDIT_FONT_ASCENT) / (2 + (value2 == NULL ? 0 : 1));
 
     // Assume the label is alpha-numeric (with a descender)
     bool onpage = PAGE_CONTAINS(baseline - (EDIT_FONT_ASCENT - 1), baseline + EDIT_FONT_DESCENT);
     if (onpage) {
       lcd_moveto(0, baseline);
-      lcd_put_u8str_P(pstr);
+      lcd_put_u8str_P(pstr1);
     }
 
-    // If a value is included, print a colon, then print the value right-justified
-    if (value != NULL) {
+    // If a value1 is included, print a colon, then print the value1 right-justified
+    if (value1 != NULL) {
       lcd_put_wchar(':');
       if (extra_row) {
-        // Assume the value is numeric (with no descender)
+        // Assume that value1 is numeric (with no descender)
         baseline += EDIT_FONT_ASCENT + 2;
         onpage = PAGE_CONTAINS(baseline - (EDIT_FONT_ASCENT - 1), baseline);
       }
       if (onpage) {
         lcd_moveto(((lcd_chr_fit - 1) - (vallen + 1)) * one_chr_width, baseline); // Right-justified, leaving padded by spaces
-        lcd_put_wchar(' '); // overwrite char if value gets shorter
-        lcd_put_u8str(value);
+        lcd_put_wchar(' '); // overwrite char if value1 gets shorter
+        lcd_put_u8str(value1);
       }
     }
+
+    #if ENABLED(BABYSTEP_DISPLAY_Z_TOTAL)
+      if (pstr2 != NULL) {
+        ui.set_font(FONT_MENU);
+        lcd_moveto(0, LCD_PIXEL_HEIGHT - MENU_FONT_DESCENT);
+        lcd_put_u8str_P(pstr2);
+        lcd_put_wchar(':');
+        lcd_put_u8str(value2);
+        lcd_moveto(LCD_PIXEL_WIDTH - 1, LCD_PIXEL_HEIGHT - 1);  // Work around a bug in u8glib that puts a random
+                                                                // ':' on the screen. This hides the ':'.
```

Comment:
@Roxy-3D — I think this was due to `value2` not being nul-terminated. I've switched to `strcpy` for setting the `tmp` string, so it should now be ok. Check to see if this workaround is still needed.

Reply:
No...    If you are interested...    Comment out that lcd_moveto() line.     There are two lcd_put_wchar(':'); in that function.   The random colon is coming from the first one.   (Not from the new code!!!!)    If you change the first one to lcd_put_wchar('=');  you start getting a random equal sign painted on the display.

It would be worth while for you to do that....    There is some lingering bug (not in this code!!!) that we probably want to find.

Reply:
Great, I figured it had to be from Marlin, not u8g. I'll make sure things are done in the correct order here.

Reply:
> Great, I figured it had to be from Marlin, not u8g. I'll make sure things are done in the correct order here.

I also investigated if it was some artifact coming from the font switching...   It isn't.   But like I say....   Comment out that last lcd_moveto() so you can see it show up.   And then change the first lcd_put_wchar('=');   and you will see a second = show up at the end of the info the new code puts out.    

It is possible it is Marlin doing that.   But if it is...   It is happening up in the menu code that is looping through things and doing the screen changes.   I don't know but I did spend some time looking for it.    The strange thing is...    some how, the u8g code is remembering what the first lcd_put_char() is putting out....   and that is what the random character is that gets painted.

Reply:
This makes me wonder if that specific `lcd_put_char` is being called twice. What if it outputs a character variable instead? `static char test = '0'; test++; lcd_put_char(test);`

Reply:
I didn't try a variable.    I would like to twist your arm hard enough that you take a quick look at it.     It would be a lot easier to figure out if I could single step through that code.   But I don't have any way to do that.

Reply:
I'll be happy to give it a run. I'm about done with the `Babystep` class now.